src/logger.ts

```diff
@@ -0,0 +1,20 @@
+import pino from 'pino';
+import type { Logger } from 'pino';
+
+const level = process.env.REACT_APP_LOG_LEVEL
+  ?? process.env.NODE_ENV === 'production' ? 'error' : 'debug';
```

Comment:
Since node won't be in production, I assume this refers to "level in a build artifact intended for production" versus "level in a build artifact for intended for development".

Does `REACT_APP_LOG_LEVEL` drive log levels for dependencies too, or only the code in our data viewer app? If it drives more, perhaps it would be better to start with a  `DATA_VIEWER_LOG_LEVEL` if it exists and fall back to `REACT_APP_LOG_LEVEL`. That way we could amplify (or mute) our app and its dependencies more easily.

Another minor note, the description says "INFO and above during development, ERROR and above in production." but the setting here appears different.

One more minor consideration, if the default level is ERROR in production, that only gives two levels to work with by default in production. It might make sense to default to WARN or INFO in production to allow one or two more levels.

Reply:
Ah, there are some React conventions here! The docs are really good: [Adding Custom Environment Variables](https://create-react-app.dev/docs/adding-custom-environment-variables/).

> Since node won't be in production, I assume this refers to "level in a build artifact intended for production" versus "level in a build artifact for intended for development".

Yep, exactly! The relevant bit about `NODE_ENV`:

> There is also a built-in environment variable called `NODE_ENV`. You can read it from `process.env.NODE_ENV`. When you run `npm start`, it is always equal to `'development'`, when you run `npm test` it is always equal to `'test'`, and when you run `npm run build` to make a production bundle, it is always equal to `'production'`.

So we can depend on that being set appropriately!

> Does `REACT_APP_LOG_LEVEL` drive log levels for dependencies too, or only the code in our data viewer app?

Right now it's just for our app; if and when I add dependencies that have configurable logging (which is relatively rare on the front-end, as you kind of allude to later), I'll try to wire them up to use the same configuration.

I'm not super familiar with Pino, and probably would not have reached for it but for its use in the back-end; I would imagine there's some degree of filtering available, but for now my plan is to just deal with that if and when we get there. 

> ... perhaps it would be better to start with a `DATA_VIEWER_LOG_LEVEL`

The other React-ism bit is that in order for a custom build-time environment variable to make it into the code, it must begin with `REACT_APP_` - this is a security measure against accidentally exposing environment variables from the build machine.

Honestly, I think this is primarily going to be helpful for development, and front-end logs are unlikely to ever be looked at or sent elsewhere.

> Another minor note, the description says "INFO and above during development, ERROR and above in production." but the setting here appears different.

Oops, yes, good catch! Thank you, I'll fix that.

> One more minor consideration, if the default level is ERROR in production, that only gives two levels to work with by default in production. It might make sense to default to WARN or INFO in production to allow one or two more levels.

You might have stronger opinions than I do about log levels! My impression is that `WARN` is for recoverable errors and deprecations and such, and that end users won't want or need to see that - is that right? Remember, this is getting logged to the browser console, so the only time folks will see this is if they open up the browser dev tools, which I hope they don't feel a need to do on a regular basis. If and when we wire up to eg DataDog or New Relic or Sentry or whatever, we would definitely consider what level we want to send at that time!

Reply:
There was also a bug on this line :sweat: which should be fixed now.

Try:

`REACT_APP_LOG_LEVEL=error npm start`, which should not show you the web-vitals log messages, or `REACT_APP_LOG_LEVEL=trace npm run build` and loading the built artifacts, which should!

Reply:
Thanks for the consideration and explanations, it all makes sense to me. I tend to prefer verbosity (plus grep) while most folks I think prefer concision. It does come down to preference for most of these decisions. It's more important that the levels are used consistently in the code of a given package than which levels to set by default.

Reply:
By the way, I found `pino` relatively familiar and straightforward to set up in the `service` project but if it is a pain or unfamiliar here I would not object to going with what works for you.